Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs

```diff
@@ -36,37 +36,49 @@ public View View
 		public void TearDown()
 		{
 			View = null;
-			_renderer?.Dispose();
-			_renderer = null;
+			Renderer?.Dispose();
+			Renderer = null;
 			_view = null;
 			_context = null;
 		}
 
-		public void LayoutView(double width, double height)
+		public void LayoutView(double width, double height, double? maxWidth = null, double? maxHeight = null)
 		{
-			_width = width;
-			_height = height;
+			if (width == -1)
+				width = double.PositiveInfinity;
+
+			if (height == -1)
+				height = double.PositiveInfinity;
+
+			Width = width;
+			Height = height;
 			Context context;
 
-			if (_renderer == null || !(_context.TryGetTarget(out context)) || !_renderer.View.IsAlive())
+			if (Renderer == null || !(_context.TryGetTarget(out context)) || !Renderer.View.IsAlive())
 				return;
 
 			if (View == null)
 			{
 				var empty = MeasureSpecFactory.GetSize(0);
-				_renderer.View.Measure(empty, empty);
+				Renderer.View.Measure(empty, empty);
 				return;
 			}
 
 			var request = View.Measure(width, height, MeasureFlags.None);
 
 			var layoutParams = NativeView.LayoutParameters;
-			if (height == -1)
+			if (height == -1 || double.IsInfinity(height))
```

Comment:
We're setting the value to `PositiveInfinite`, do we need the check for `-1`?

Reply:
👍